app/controllers.js

```diff
@@ -610,6 +611,70 @@ Controller.prototype.handleApiEditProject = function(req, res) {
     });
 };
 
+// Create or overwrite an entry in phase history
+Controller.prototype.handleApiUpdatePhaseHistory = function(req, res) {
+    projectCache.getById(req.params.projectId, function(error, project) {
+        if(!project) {
+            log.debug('Error updating project. Project with given ID was not found.');
+            apiUtils.handleResultSet(res, 422,
+                new ApiResponse(null, ['Error updating project. Project with given ID was not found.'])
+            );
+        } else {
+            project.setPhaseHistoryEntry(req.body.phase, req.body.label, req.body.month, req.body.year);
+
+            projectDao.addProject(project, function(err, editedProject) {
+                log.trace('Updating project history: ' + JSON.stringify(project));
```

Comment:
General rule: we should avoid dumping all the potentially sensitive data even if it is a trace log entry - someone misconfiguring the server with wrong log settings could leak that to a log monitoring service